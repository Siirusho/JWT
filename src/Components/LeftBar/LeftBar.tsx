import { Container, makeStyles, Typography, Theme, createStyles } from "@material-ui/core";
import {
    Home,
    Person,
    PhotoCamera,
    PlayCircleOutline,
    List,
    Devices,
    Collections,
    Settings,
    ExitToApp
} from "@material-ui/icons";
import { FC } from 'react';

const useStyles = makeStyles<Theme>(theme =>
    createStyles({
        item: {
            display: "flex",
            alignItems: "center",
            marginBottom: theme.spacing(4),
            [theme.breakpoints.up("sm")]: {
                marginBottom: theme.spacing(3),
                cursor: "pointer",
            }
        },
        icon: {
            marginRight: theme.spacing(1),
            [theme.breakpoints.up("sm")]: {
                fontSize: "18px",
            }
        },
        text: {
            [theme.breakpoints.down("sm")]: {
                display: "none"
            }
        },
        container: {
            height: "100vh",
            paddingTop: theme.spacing(10),
            backgroundColor: theme.palette.primary.main,
            color: "white",
            position: "sticky",
            top: 0, left: 0,
            [theme.breakpoints.up("sm")]: {
                backgroundColor: "white",
                color: "#555",
            }
        }
    })
);


const LeftBar: FC = () => {
    const classes = useStyles();
    const navBar = [
        { id: 1, component: <Home className={classes.icon} />, label: 'HomePage' },
        { id: 2, component: <Person className={classes.icon} />, label: "Friends" },
        { id: 3, component: <List className={classes.icon} />, label: "List" },
        { id: 4, component: <PlayCircleOutline className={classes.icon} />, label: "Music" },
        { id: 5, component: <PhotoCamera className={classes.icon} />, label: "Videos" },
        { id: 6, component: <Devices className={classes.icon} />, label: "Apps" },
        { id: 7, component: <Collections className={classes.icon} />, label: "Collections" },
        { id: 8, component: <Settings className={classes.icon} />, label: "Settings" },
        { id: 9, component: <ExitToApp className={classes.icon} />, label: "Log Out" },
    ]
    return (
        <Container className={classes.container}>
            {
                navBar.map((item) => {
                    return (
                        <div className={classes.item} key={item.id}>
                            {item.component}
                            <Typography className={classes.text}>{item.label}</Typography>
                        </div>
                    )
                })
            }
        </Container>
    );
}

export default LeftBar;