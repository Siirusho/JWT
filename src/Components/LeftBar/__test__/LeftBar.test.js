import React from "react";
import LeftBar from "../LeftBar";
import { render } from "@testing-library/react";


describe("Left side bar", () => {
    it("should render correct LeftBar component and match with snapshot", () => {
        const { asFragment } = render(<LeftBar />);
        expect(asFragment(<LeftBar />)).toMatchSnapshot();
    })
});