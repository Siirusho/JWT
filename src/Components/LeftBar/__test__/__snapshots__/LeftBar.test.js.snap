// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left side bar should render correct LeftBar component and match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-container-4 MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        HomePage
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Friends
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        List
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Music
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="12"
          cy="12"
          r="3.2"
        />
        <path
          d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Videos
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 6h18V4H4c-1.1 0-2 .9-2 2v11H0v3h14v-3H4V6zm19 2h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 9h-4v-7h4v7z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Apps
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Collections
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Settings
      </p>
    </div>
    <div
      class="makeStyles-item-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-text-3 MuiTypography-body1"
      >
        Log Out
      </p>
    </div>
  </div>
</DocumentFragment>
`;
